JS

```diff
@@ -0,0 +1,211 @@
+
+function Hamburger(type, count,stuffing){
+  this.name = this[type].name;
+  this.count = count;
+  this.stuffingName = this[stuffing].name;
+  this.price = this[type].price + this[stuffing].price;
+  this.calories = this[type].calories + this[stuffing].calories;
+}
+
+Hamburger.prototype.LARGE_SIZE= { 
+  name:"Big hamburger",
+  price: 100,
+  calories: 40
+};
+
+Hamburger.prototype.SMALL_SIZE= {
+  name:"Small hamburger",
+  price: 50,
+  calories: 20
+  };
+
+Hamburger.prototype.STUFFING_CHEESE = {
+  name: 'Cheese',
+  price: 10,
+  calories: 20
+};
+
+Hamburger.prototype.STUFFING_SALAD = {
+  name: 'Salad',
+  price: 20,
+  calories: 5
+};
+
+Hamburger.prototype.STUFFING_POTATO = {
+  name: 'Potato',
+  price: 15,
+  calories: 10
+};
+
+Hamburger.prototype.calculatePrice = function(){
+  return this.price*this.count;
+}
+
+Hamburger.prototype.calculateCalories = function(){
+  return this.calories*this.count;
+}
+
+Hamburger.prototype.getName = function(){
+  return this.name +' '+ this.stuffingName;
+}
+
+Hamburger.prototype.getCount = function(){
+  return this.count;
+}
+
+function SaladAndDrink(type, count){
```

Comment:
@Mikleprog Actually it's better to split classes for Salad and Drink since they are different types